File: LeetCode_Marathons/LCM3/Easy/Valid_Mountain_Array/a.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        int n = arr.size();
        bool isIncreasing = true;
        for (int i = 0; i < n - 1; i++) {
            if (isIncreasing) {
                if (arr[i + 1] < arr[i]) {
                    if (i != 0) isIncreasing = false;
                    else return false;
                } else if (arr[i + 1] == arr[i]) return false;
            } else {
                if (arr[i + 1] >= arr[i]) return false;
            }
        }
        return !isIncreasing;
    }
};

File: LeetCode_Marathons/LCM3/Easy/Valid_Mountain_Array/a.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        n: int = len(arr)
        if n < 3: return False
        isIncreasing: bool = True
        for i in range(n - 1):
            if isIncreasing:
                if arr[i + 1] < arr[i]:
                    if i != 0: isIncreasing = False
                    else: return False
                elif arr[i + 1] == arr[i]: return False
            else:
                if arr[i + 1] >= arr[i]: return False
        return not isIncreasing